templates/history.sh.erb

```diff
@@ -1,10 +1,16 @@
 # This file is managed by Puppet.
 export HISTTIMEFORMAT="%h %d %H:%M:%S "
 export HISTSIZE=<%= @size %>
+PARENT_USER=$(ps -o user= $(ps -o ppid= $(ps -o ppid= $PPID)))
 if [ -n "$SUDO_USER" ]; then
   export HISTFILE=~/.bash_${SUDO_USER}_<%= @filename %>
+elif [ "$USER" == "root" -a "$USER" != "$PARENT_USER" ]; then
```

Comment:
Perhaps this should change to:
```suggestion
elif [ "$PARENT_USER" != "root" -a "$USER" != "$PARENT_USER" ]; then
```
I think that might support if a user runs `sudo su - USERNAME` to sudo to a non-root user.